compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -478,6 +482,10 @@ class ExpressionCodegen(
             val parameterType = callable.valueParameterTypes[i]
             require(arg != null) { "Null argument in ExpressionCodegen for parameter ${irParameter.render()}" }
             callGenerator.genValueAndPut(irParameter, arg, parameterType, this, data)
+
+            if (expression.symbol.owner.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA && arg is IrConst<*>) {
```

Comment:
`IrConst<*>` expression is just a constant literal. It doesn't imply anything about given call expression being a result of direct invoke optimization for a lambda.